### PR TITLE
Open registration

### PR DESCRIPTION
**Pull request by plytro**
We'll do it live! Not really though, someone needs to merge and deploy this for that to actually happen.

jerry 2012 - https://thismight.be/offensive/?c=comments&fileid=343039
komara 2015 - https://thismight.be/offensive/?c=comments&fileid=391612
kapuski 2015 - https://thismight.be/offensive/?c=comments&fileid=395389

I recall others but I didn't want to search too hard.

### CODE REVIEW

**Comment by numist** [comment]
This is something I've been pondering on and off over the last few years.

I think opening registration is a good idea, but it should be better regulated than "registration is open" vs "invitation only". A good metering rate might allow registration only if the most recently-opened account is at least a day (or few?) old.

If someone loses the race to POST their registration, it could render "sorry, someone beat you to it. try again in a couple of days."

The non-invite registration flow hasn't been tested since 2006, so we'll also need to test this  exhaustively in a dev environment to ensure the rest of the site hasn't diverged too much from what `registr` expects.

Paging @themaxx for sanity review.

**Comment by themaxx** [comment]
As you probably already know, the primary—and possibly only—motivation for an invite-only system was to make banishment work. It's impossible—as far as I can tell—to prevent bussed people from creating new accounts. Can't block them by IP, or with cookies or local storage, etc.. 

If someone knows how to solve that problem I'm comfortable with at least trying out open registration.

**Comment by themaxx** [comment]
I'm a little confused about how hiding the referral link input accomplishes this.

**Comment by themaxx** [comment]
derp.

**Comment by plytro** [comment]
We could open it up, after 500 or 1000 new users register, close it back down again and sort out the cruft. It would require the least amount of coding.

**Comment by numist** [comment]
I do not look forward to what my life would look like as we "sorted out the cruft"…

**Comment by plytro** [comment]
Good point I don't get the emails or everyone asking me to fix anything

--Steve

**Comment by ssx** [comment]
Invite only in my opinion. Our community would dilute massively if it was a free for all.

**Comment by plytro** [comment]
The overall problem is the community is dying because people have exhausted their invite lists. Our active contributor size is about 12-20 people daily. That number continues to shrink. I would be interested to see what a query of select count(distinct userid), date(timestamp) from offensive_uploads group by date(timestamp) order date(timestamp) desc; shows. I would wager it is a steady downward trend with a few plateaus and a couple of spikes such as IRLARD

**Comment by numist** [comment]
Let's get a change in that meters the incoming rate to a max of one new walk-up account per day and see how it goes.

File: offensive/registr.php [hunk 1]
```diff
@@ -64,6 +64,9 @@ function handleAccountRequest() {
 	
 	function getReferrerId( $refcode ) {
 
+    // open up registration
+    return 1;
```
Comment (numist) [inline offensive/registr.php]: Returning before running the query would break normal referral attribution.

**Comment by numist** [comment]
This is superceded by #101.

**Pull request closed without merging**